src/path.js

```diff
@@ -484,7 +484,7 @@ class FieldPath extends Path {
   static validateFieldPath(fieldPath) {
     if (!is.instanceof(fieldPath, FieldPath)) {
       if (!is.string(fieldPath)) {
-        throw new Error(`Paths must be strings or FieldPath objects.`);
+        validate.throwCustomObjectError(fieldPath);
```

Comment:
FYI- I think this means if you did something like `.update(3, ...)` you'd get an error saying "Cannot use custom type 'number' as a Firestore type." which is a worse error message than before.  I don't care too much though.

Reply:
This ends up being: "Update() requires either a single JavaScript object or an alternating list of field/value pairs that can be followed by an optional precondition. Argument at index 3 is not a valid FieldPath. Cannot use custom type "Object" as a Firestore type."